eng/validation/build-arcadewithrepo.ps1

```diff
@@ -195,52 +210,105 @@ function Git-Command($repoName) {
+    $branchExists = Git-Command $githubRepoName ls-remote --heads $azdoUri refs/heads/$targetBranch
+}
 if($null -ne $branchExists)
 {
+    Write-Host "${targetBranch} was found. Attempting to clean up."
     try
     {
         & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
-        Git-Command $targetRepoName push origin --delete $targetBranch
+        if($true -eq $pushBranchToGithub)
+        {
+            & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcGitHubRepoName --github-pat $githubPAT --password $barToken
+            Git-Command $githubRepoName push origin --delete $targetBranch
+        }
+        else
+        {
+            & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcAzDORepoName --azdev-pat $azdoToken --password $barToken
+            Git-Command $githubRepoName remote add $remoteName $azdoUri
+            Git-Command $githubRepoName push $remoteName --delete $targetBranch
+        }
     }
     catch
     {
-        Write-Warning "Unable to delete default or branch when cleaning up existing branch"
+        Write-Warning "Unable to delete default channel or branch when cleaning up existing branch"
     }
 }
-Git-Command $targetRepoName checkout -b $targetBranch $sha
+
+## Create a branch from the repo with the given SHA.
+Git-Command $githubRepoName checkout -b $targetBranch $sha
 
 ## Make the changes to that branch to update Arcade - use darc
-Set-Location $(Get-Repo-Location $targetRepoName)
-& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --password $barToken
+Set-Location $(Get-Repo-Location $githubRepoName)
+& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --azdev-pat $azdoToken --password $barToken
 
-## Push branch to github
-Git-Command $targetRepoName commit -am "Arcade Validation test branch - version ${arcadeSdkVersion}"
-Git-Command $targetRepoName push origin HEAD
+Git-Command $githubRepoName commit -am "Arcade Validation test branch - version ${arcadeSdkVersion}"
 
-## Push branch to AzDO org/project with the official pipeline to build the repo
-## Don't need to do this for Roslyn, but we'll need to do it for Runtime. 
+if($true -eq $pushBranchToGithub)
+{
+    ## Push branch to github
+    Git-Command $githubRepoName push origin HEAD
 
-## Add default channel from that github repo and branch to "General Testing"
-& darc add-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
+    ## Add default channel from that github repo and branch to "General Testing"
+    & darc add-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcGitHubRepoName --github-pat $githubPAT --password $barToken
```

Comment:
[minor nit] You can take this and LN 299 out of the if so that there's only one darc command. That way if we need to tweak it we only need to do it in one place.

Reply:
Would have the same advantage during deletion, you can skip that whole if.

Reply:
These commands are taking in different parameters though...

Reply:
Can you populate a $darcRepoName variable with the Github or azdo variable inside the if? Is there another difference?

Reply:
> You're actually downgrading the version and hence you got some merge conflicts. You can run another update-dependencies from the validation channel and your branch should be good.

Okay, I think it's where it needs to be. :)

Reply:
That works better. I have the conditional assign a $darcRepoName and then use that in the darc command outside the conditional. Also including both GitHub and AzDO tokens in the darc call.